### PR TITLE
Update devDependencies and downgrade eslint-plugin-promise

### PR DESCRIPTION
**Pull request by XhmikosR**
eslint-config-semistandard requires eslint-plugin-promise@5

FYI @Trott @nschonni so that you don't land the update in the future.

Unsure why things worked but if I remove the lock file locally I get the errors as expected:

```
C:\Users\xmr\Desktop\nodejs.org>npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: nodejs.org@undefined
npm ERR! Found: eslint-plugin-promise@6.0.0
npm ERR! node_modules/eslint-plugin-promise
npm ERR!   dev eslint-plugin-promise@"^6.0.0" from the root project
npm ERR!   peer eslint-plugin-promise@">=4.2.1" from eslint-config-semistandard@16.0.0
npm ERR!   node_modules/eslint-config-semistandard
npm ERR!     dev eslint-config-semistandard@"^16.0.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint-plugin-promise@"^4.2.1 || ^5.0.0" from eslint-config-standard@16.0.3
npm ERR! node_modules/eslint-config-standard
npm ERR!   peer eslint-config-standard@">=16.0.3" from eslint-config-semistandard@16.0.0
npm ERR!   node_modules/eslint-config-semistandard
npm ERR!     dev eslint-config-semistandard@"^16.0.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\xmr\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\xmr\AppData\Local\npm-cache\_logs\2021-12-23T07_23_56_061Z-debug.log
```

### CODE REVIEW

**Comment by Trott** [comment]
> Unsure why things worked but if I remove the lock file locally I get the errors as expected:

Maybe the lock file was generated with npm@6 but you're using npm@7 or npm@8? Maybe it will work if you do `npm i --legacy-peer-deps`?

**Comment by XhmikosR** [comment]
But the idea isn't to bypass the error but keep using compatible versions.

BTW, I started getting notifications for each PR on this repo while
previously I didn't get any.

**Review by Trott — APPROVED** [review]

**Comment by Trott** [comment]
> BTW, I started getting notifications for each PR on this repo while previously I didn't get any.

I have no idea what would have caused that, but I also can't completely rule out that I didn't do something somewhere somehow to make that happen.

**Comment by XhmikosR** [comment]
It must have something to do with Code owners. Unsure how I can skip getting the PR notifications, though. I'm getting a handful of notifications from the various projects I contribute and it's wearing me off :/

File: package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "eslint-plugin-markdown": "^2.2.1",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^4.0.0",
-    "eslint-plugin-promise": "^6.0.0",
```
Comment (m0dul3) [inline package.json:70]: #4333

**Pull request merged**